data_and_models/pipelines/ner/preprocess.py

```diff
@@ -0,0 +1,93 @@
+# Adapted from https://github.com/explosion/projects/blob/v3/tutorials/ner_drugs/scripts/preprocess.py.
```

Comment:
It would indeed be better but be cannot with what has been done before.

Indeed, we used an undocumented and non standard export to `JSONL` instead of using `prodigy data-to-spacy`.
We therefore ended up with a file format which cannot be automatically used by `spacy convert`.

Reply:
@EmilieDel 

Yes.

As said in the out of scope section of the TODO list (https://github.com/BlueBrain/Search/issues/274#issuecomment-801108330), there is an issue for it:

> Converting permanently annotations from .jsonl to .spacy and removing the dedicated DVC stages. See #308.

Reply:
> Indeed, we used an home-made export to JSONL instead of using `prodigy data-to-spacy`. We therefore ended up with a file format which cannot be automatically used by spacy convert.

@pafonta can you please elaborate on this? Iirc the JSONL are dumps generated using the standard [`db-out`](https://prodi.gy/docs/recipes#db-out), we didn't use any home-made solution?

Reply:
The `JSONL` files from `prodigy db-out` are not standard in the environment of `spaCy`.
The JSON files from `prodigy data-to-spacy` are however standard.

What I was trying to say is not that we did a home-made code to export annotations.
I was trying to say that we used a non standard way to export annotations while being in a `spaCy` environment.

So, as we didn't follow the standard way we cannot use the standard tools and workflows.

Reply:
spacy2 can convert JSONL to JSON
```
spacy convert --lang en my.jsonl .
```
would that be an option?

Reply:
The current plan is to convert on-the-fly `.jsonl` files to `.spacy` files until #308 is merged.
This is what `preprocess.py` does.

All other approaches, like using `Prodigy` or `spaCy 2` have the blocker of requiring to either:

1. Keep the old stack (`Prodigy` or `spaCy 2`) aside the new one (`no Prodigy` and `spaCy 3`).
2. Have the annotations both in `.jsonl` files for `eval.py` and in `.json` and/or `.spacy` files for `spaCy 3`.

Point 1 is a nightmare to maintain and add a lot of complexity compared to `preprocess.py`.
For point 2, it was decided to **not** do so between @FrancescoCasalegno, @EmilieDel, and I.

So, unless I miss something, I would say `preprocess.py` is still the most appropriate proposal.

Reply:
It looks like this is the best solution.
This thread can be closed for me.